src/serde_helpers.rs

```diff
@@ -289,20 +289,20 @@ pub mod hex_string_as_object_id {
     }
 }
 
-/// Contains functions to serialize a Uuid as a bson::Binary and deserialize a Uuid from a
-/// bson::Binary.
+/// Contains functions to serialize a [`uuid::Uuid`] as a [`bson::Binary`] and deserialize a
+/// [`uuid::Uuid`] from a [`bson::Binary`]. This only works with version 0.8 of the [`uuid`] crate.
 ///
 /// ```rust
 /// # use serde::{Serialize, Deserialize};
 /// # use uuid::Uuid;
-/// # use bson::serde_helpers::uuid_as_binary;
+/// # use bson::serde_helpers::uuid_0_8_as_binary;
 /// #[derive(Serialize, Deserialize)]
 /// struct Item {
-///     #[serde(with = "uuid_as_binary")]
+///     #[serde(with = "uuid_0_8_as_binary")]
 ///     pub id: Uuid,
 /// }
 /// ```
-pub mod uuid_as_binary {
+pub mod uuid_0_8_as_binary {
```

Comment:
Note that with #239, each major version of `uuid` will come with 4 serde helper modules, which will pollute the namespace a bit. That doesn't seem too bad to me but we could consider gating it behind a feature flag if we want to avoid that.

Reply:
I don't feel very strongly either way

Reply:
Yeah I don't think that's a huge deal, I'm fine leaving as-is.